### DISCUSSION TITLE
Select widget with default value already selected

### DISCUSSION THREAD

**Discussion by SkwalExe** [Textualize/textual#3381] [Q&A] [answered]
I don't think it is possible yet to have a Select with a default value already selected when the widget is mounted, would it be a good idea ? 

Should I create a pr?

**Comment by willmcgugan** [comment, marked as answer]
You can set the `value` parameter on the [constructor](https://textual.textualize.io/widgets/select/#textual.widgets.Select)

**Comment by SkwalExe** [comment]
Isn't it weird that you must set the `value` parameter to the value of the default option and not its "text" ? 

Imagine having for example a list of countries with data associated to it ( an option would be `("country_name", data)`), you would want to specify the default option by the name of the country and not by its associated data right, am I missing something here?

**Reply by davep** [reply]
What you're calling the data is probably better thought of as an ID. So a Select's options is a collection of values with readable descriptions. In the case of countries, for example, you'd have country codes and names. 

It's reasonable to set the default with the code, not the name (which could potentially be coming via a translation library).

**Comment by SkwalExe** [comment]
Thank you, now I understand. Also, is it normal that when the `value` property is accessed the default value doesn't work?

```python
from textual.widgets import Select
from textual.app import App

def do_nothing(x):
    pass

class Test(App):
    def compose(self):
        select =  Select(options=[("France", "fr")], value="fr")
        # When the line below is executed, the select widget doesn't have any default value but it should
        do_nothing(select.value)
        yield select

app = Test()
app.run()
```

**Reply by davep** [reply]
If I recall correctly the value reactive won't be populated until the DOM is loaded; so from on_ready onwards.
